modules/docker/Makefile.build

```diff
@@ -0,0 +1,43 @@
+## Use DOCKER_IMAGE_NAME envvar to specify docker image with tags
+## User ARGS to pass arguments
+DOCKER_BUILD_PATH ?= .
+DOCKER_BUILD_FLAGS ?= --no-cache
```

Comment:
Why would you want the `--no-cache` option as a default? Our other repos don't use this option.

I would think we'd want to use the docker build cache 99% of the time.

I think a better argument could be made for using `--pull` as the one default flag.

Reply:
Another follow-up:

I spent a fair bit of work trying to ensure our docker images build quickly in CI using the `--cache-from` option to make sure we pull a relevant image cache and avoid unnecessarily re-building image layers.
https://github.com/oneconcern/built-object-service/blob/623ca520f8f11ad2e3d30101fd7b330e8d5d0ed0/hack/make/docker.mk#L47-L55

This seems to throw away all that which feels like a step backwards to me.